### PR TITLE
Fix LLMM1 kernel

### PR DESCRIPTION
**Pull request by fxmarty**
As per title.

The kernel is broken and yields wrong generation, for example with TP=2 & llama 2 7B. There are two main issues in the kernel:
* Number of thread in a block is not a multiple of the warp size, although we use warp shuffle operations.
* Indexing on the block of rows is wrong & indexing to load the weight matrix is wrong.

Example repro:
```python
import torch
from transformers import AutoModelForCausalLM
# from gemv_ext import LLMM1
from vllm import _custom_C

m = 4096
n = 1

for k in range(200, 1600, 8):
    with torch.no_grad():

        down_proj_weight = torch.rand(m, k, device="cuda", dtype=torch.float16)
        inp = torch.rand(n, k, device="cuda", dtype=torch.float16) - 0.5

        out_functional = torch.nn.functional.linear(inp, down_proj_weight)

        out_custom = torch.empty(n, m, dtype=inp.dtype, device="cuda")
        _custom_C.LLMM1(down_proj_weight, inp, out_custom, 4)

        absdiff = (out_functional.to(torch.float32) - out_custom.to(torch.float32)).abs()

        print("------ k:", k)
        print("  % off:", absdiff.mean().item() / out_functional.abs().mean().item())
```

### CODE REVIEW

**Review by dllehr-amd — APPROVED** [review]
@fxmarty I appreciate the PR!  We were mainly focused on TP=8 and TP=1.  These changes look good to me.

**Comment by fxmarty** [comment]
Thank you, I fixed the linting conflict.

**Comment by shajrawi** [comment]
Still have failing CI

Run codespell --toml pyproject.toml
./csrc/custom/custom_kernels.cu:88: accross ==> across
Error: Process completed with exit code 65.

**Comment by fxmarty** [comment]
@shajrawi thank you, should be fixed

**Comment by shajrawi** [comment]
Merged! Thanks for the fix!!
